### PR TITLE
CVEs alerts inventory for Vulnerability Detector - VDT and WDB Integration Tests implementation

### PR DESCRIPTION
**Pull request by DProvinciani**
|Related issue|
|---|
| https://github.com/wazuh/wazuh-qa/issues/1242 |

## Description

This pull request includes all the necessary test cases to cover the new Wazuh DB commands and Vulnerability Detector alerting mechanisms created as part of the epic https://github.com/wazuh/wazuh/issues/7749. For details, check the description of the issue https://github.com/wazuh/wazuh-qa/issues/1242 and https://github.com/wazuh/wazuh-qa/issues/1261.

**Vulnerability Detector**:
- test_scan_results: Includes tests that check vulnerabilities insertion and removal in: 
  - Log file
  - Alerts file
  - Vulnerabilities inventory database. 
- test_scan_types: Includes tests for **Baseline**, **Partial** and **Full** scans
- test_general_settings: Includes new settings **min_full_scan_interval** and **retry_interval**

**Wazuh DB**:
- The set of tests implemented verify the right behavior of all the new Wazuh DB commands implemented as part of the epic https://github.com/wazuh/wazuh/issues/7749.

## Dod
`test_scan_types`

types

`test_general_settings`

general_settings

`test_scan_results`

scan_results

`test_windows`

image

`tests_wazuh_db`

image

## Tests

- [X] Proven that tests **pass** when they have to pass.
- [X] Proven that tests **fail** when they have to fail.

- [X] Python codebase satisfies PEP-8 style style guide. `pycodestyle --max-line-length=120 --show-source --show-pep8 file.py`.
- [X] Python codebase is documented following the Google Style for Python docstrings.
- [X] The test is documented in wazuh-qa/docs.

### CODE REVIEW

**Review by BraulioV — APPROVED** [review]

**Comment by DProvinciani** [comment]
### Jenkins execution

https://ci.wazuh.info/job/Test_integration/6716/

image

File: tests/integration/test_wazuh_db/data/agent_messages.yaml [hunk 1]
```diff
@@ -1,77 +1,300 @@
...
+    output: ['ok {"action":"UPDATE","status":"SUCCESS"}']
+    stage: "agent vuln_cves update already inserted entry"
   -
-    input: 'agent 001 vuln_cve insert {"name":"test package","version":"1.0","architecture":"x86","cve":"1001"}'
-    output: "ok"
-    stage: "agent vuln_cve insert with spaces in json payload"
+    input: 'agent 000 sql INSERT INTO sys_programs (scan_id,scan_time,format,name,priority,section,size,vendor,install_time,version,architecture,multiarch,source,description,location,triaged,cpe,msu_name,checksum,item_id)
+                                             VALUES(0,"2021/04/07 22:00:00","deb","test package","optional","utils","7490","Wazuh wazuh@wazuh.com",NULL,"1.0.0","amd64",NULL,NULL,"Test package",NULL,0,NULL,NULL,"e7dbc9bba5a0ee252866536225b952d3de7ea5cb","777fef8cc434b597769d102361af718d29ef72c1")'
+    output: ['ok []']
+    stage: "agent vuln_cves adding dummy test package to sys_programs"
```
Comment (juliamagan) [inline tests/integration/test_wazuh_db/data/agent_messages.yaml:47]: Same as above, we need to change the quotes.

File: tests/integration/test_wazuh_db/data/agent_messages.yaml [hunk 2]
```diff
@@ -1,77 +1,300 @@
...
   -
-    input: 'agent 000 vuln_cve clear'
-    output: "ok"
-    stage: "agent vuln_cve clear table"
+    input: 'agent 000 sql SELECT * FROM sys_programs WHERE name = "test package"'
+    output: ['ok [{"scan_id":0,"scan_time":"2021/04/07 22:00:00","format":"deb","name":"test package","priority":"optional","section":"utils","size":7490,"vendor":"Wazuh wazuh@wazuh.com","version":"1.0.0","architecture":"amd64","description":"Test package","triaged":0,"checksum":"e7dbc9bba5a0ee252866536225b952d3de7ea5cb","item_id":"777fef8cc434b597769d102361af718d29ef72c1"}]']
```
Comment (juliamagan) [inline tests/integration/test_wazuh_db/data/agent_messages.yaml:50]: Same as above, we need to change the quotes.

File: tests/integration/test_wazuh_db/data/agent_messages.yaml [hunk 3]
```diff
@@ -1,77 +1,300 @@
...
+    stage: "agent vuln_cves checking test package in sys_programs"
```
Comment (juliamagan) [inline tests/integration/test_wazuh_db/data/agent_messages.yaml:51]: Same as above, we need to change the quotes.

File: tests/integration/test_wazuh_db/data/agent_messages.yaml [hunk 4]
```diff
@@ -1,77 +1,300 @@
...
+    input: 'agent 000 vuln_cves insert {"name":"test package",
+                                        "version":"1.0",
+                                        "architecture":"x86",
+                                        "cve":"CVE-2021-1002",
+                                        "reference":"777fef8cc434b597769d102361af718d29ef72c1",
+                                        "type":"OS",
+                                        "status":"PENDING",
+                                        "check_pkg_existence":true}'
+    output: ['ok {"action":"INSERT","status":"SUCCESS"}']
+    stage: "agent vuln_cves insert with spaces in json payload and the test package exist in sys_programs"
```
Comment (juliamagan) [inline tests/integration/test_wazuh_db/data/agent_messages.yaml:62]: Same as above, we need to change the quotes.

File: tests/integration/test_wazuh_db/data/agent_messages.yaml [hunk 5]
```diff
@@ -1,77 +1,300 @@
...
   -
-    input: 'agent 000 vuln_cve insert {"name":"test_package","cve":"CVE-2021-1001"}'
-    output: "err Invalid JSON data, missing required fields"
-    stage: "agent vuln_cve insert incomplete package"
+    input: 'agent 000 sql SELECT * FROM vuln_cves WHERE name = "test package"'
+    output: ['ok [{"name":"test package","version":"1.0","architecture":"x86","cve":"CVE-2021-1002","reference":"777fef8cc434b597769d102361af718d29ef72c1","type":"OS","status":"PENDING"}]']
+    stage: "agent vuln_cves checking test package"
```
Comment (juliamagan) [inline tests/integration/test_wazuh_db/data/agent_messages.yaml:66]: Same as above, we need to change the quotes.

File: tests/integration/test_wazuh_db/data/agent_messages.yaml [hunk 6]
```diff
@@ -1,77 +1,300 @@
...
   -
-    input: 'agent 000 vuln_cve insert {"name":"test_package",'
-    output: "err Invalid JSON syntax, near '{\"name\":\"test_package\",'"
-    stage: "agent vuln_cve insert invalid JSON"
+    input: 'agent 000 vuln_cves insert {"name":"test_package","cve":"CVE-2021-1001"}'
+    output: ["err Invalid JSON data, missing required fields"]
+    stage: "agent vuln_cves insert incomplete package"
```
Comment (juliamagan) [inline tests/integration/test_wazuh_db/data/agent_messages.yaml:70]: Same as above, we need to change the quotes.

File: tests/integration/test_wazuh_db/data/agent_messages.yaml [hunk 7]
```diff
@@ -1,77 +1,300 @@
...
   -
-    input: 'agent 000 vuln_cve'
-    output: "err Invalid vuln_cve query syntax, near 'vuln_cve'"
-    stage: "agent vuln_cve missing command"
+    input: 'agent 000 vuln_cves insert {"name":"test_package",'
```
Comment (juliamagan) [inline tests/integration/test_wazuh_db/data/agent_messages.yaml:72]: Same as above, we need to change the quotes.

File: tests/integration/test_wazuh_db/data/agent_messages.yaml [hunk 8]
```diff
@@ -1,77 +1,300 @@
...
+    output: ["err Invalid JSON syntax, near '{\"name\":\"test_package\",'"]
+    stage: "agent vuln_cves insert invalid JSON"
```
Comment (juliamagan) [inline tests/integration/test_wazuh_db/data/agent_messages.yaml:74]: Same as above, we need to change the quotes.

File: tests/integration/test_wazuh_db/test_wazuh_db.py [hunk 9]
```diff
@@ -76,6 +77,36 @@ def pre_insert_agents():
         assert data[0] == 'ok', f'Unable to update agent {id}'
 
 
+@pytest.fixture(scope="function")
```
Comment (juliamagan) [inline tests/integration/test_wazuh_db/test_wazuh_db.py:80]:
```suggestion
@pytest.fixture(scope='function')
```

File: tests/integration/test_wazuh_db/test_wazuh_db.py [hunk 10]
```diff
@@ -76,6 +77,36 @@ def pre_insert_agents():
...
+def pre_set_sync_info():
+    """Asign the last_attempt value to last_completion in sync_info table to force the synced status"""
+
+    command = 'agent 000 sql UPDATE sync_info SET last_completion = 10, last_attempt = 10 ' \
+              'where component = "syscollector-packages"'
```
Comment (juliamagan) [inline tests/integration/test_wazuh_db/test_wazuh_db.py:85]: Same as above, we need to change the quotes.

File: tests/integration/test_wazuh_db/test_wazuh_db.py [hunk 11]
```diff
@@ -76,6 +77,36 @@ def pre_insert_agents():
...
+    receiver_sockets[0].send(command, size=True)
+    response = receiver_sockets[0].receive(size=True).decode()
+    data = response.split(" ", 1)
```
Comment (juliamagan) [inline tests/integration/test_wazuh_db/test_wazuh_db.py:88]:
```suggestion
    data = response.split(' ', 1)
```
For `split`, the default separator is any whitespace. If it is possible, we should use:

```suggestion
    data = response.split(1)
```

File: tests/integration/test_wazuh_db/test_wazuh_db.py [hunk 12]
```diff
@@ -76,6 +77,36 @@ def pre_insert_agents():
...
+    assert data[0] == 'ok', 'Unable to set sync_info table'
+
+
+@pytest.fixture(scope="function")
```
Comment (juliamagan) [inline tests/integration/test_wazuh_db/test_wazuh_db.py:92]:
```suggestion
@pytest.fixture(scope='function')
```

File: tests/integration/test_wazuh_db/test_wazuh_db.py [hunk 13]
```diff
@@ -76,6 +77,36 @@ def pre_insert_agents():
...
+    """Insert a set of dummy packages into sys_programs table"""
+
+    PACKAGES_NUMBER = 20000
+    for pkg_n in range(PACKAGES_NUMBER):
+        command = f'agent 000 sql INSERT OR REPLACE INTO sys_programs \
+        (scan_id,scan_time,format,name,priority,section,size,vendor,install_time,version,\
+        architecture,multiarch,source,description,location,triaged,cpe,msu_name,checksum,item_id)\
+        VALUES(0,"2021/04/07 22:00:00","deb","test_package_{pkg_n}","optional","utils",{random.randint(200,1000)},\
+        "Wazuh wazuh@wazuh.com",NULL,"{random.randint(1,10)}.0.0","all",NULL,NULL,"Test package {pkg_n}",\
+        NULL,0,NULL,NULL,"{random.getrandbits(128)}","{random.getrandbits(128)}")'
```
Comment (juliamagan) [inline tests/integration/test_wazuh_db/test_wazuh_db.py:103]: We need to change simple quotes to double quotes and vice versa.

File: tests/integration/test_wazuh_db/test_wazuh_db.py [hunk 14]
```diff
@@ -76,6 +77,36 @@ def pre_insert_agents():
...
+        receiver_sockets[0].send(command, size=True)
+        response = receiver_sockets[0].receive(size=True).decode()
+        data = response.split(" ", 1)
```
Comment (juliamagan) [inline tests/integration/test_wazuh_db/test_wazuh_db.py:106]: Same as above, split uses whitespaces by default.

File: tests/integration/test_wazuh_db/test_wazuh_db.py [hunk 15]
```diff
@@ -76,6 +77,36 @@ def pre_insert_agents():
...
+        assert data[0] == 'ok', f'Unable to insert package {pkg_n}'
```
Comment (juliamagan) [inline tests/integration/test_wazuh_db/test_wazuh_db.py:107]:
```suggestion
        assert data[0] == 'ok', f"Unable to insert package {pkg_n}"
```

File: tests/integration/test_wazuh_db/test_wazuh_db.py [hunk 16]
```diff
@@ -112,28 +142,32 @@ def test_wazuh_db_create_agent(configure_sockets_environment, connect_to_sockets
...
-        response = receiver_sockets[0].receive(size=True).decode()
 
+
+def test_wazuh_db_timeout(configure_sockets_environment, connect_to_sockets_module,
+                          pre_insert_packages, pre_set_sync_info):
+    """Check that effectively the socket is closed after timeout is reached"""
+
+    command = 'agent 000 package get'
+    receiver_sockets[0].send(command, size=True)
+    time.sleep(2)
```
Comment (juliamagan) [inline tests/integration/test_wazuh_db/test_wazuh_db.py:160]: It is needed to create a variable with an explanation of why the sleep of 2 seconds is needed.

File: tests/integration/test_wazuh_db/test_wazuh_db.py [hunk 17]
```diff
@@ -112,28 +142,32 @@ def test_wazuh_db_create_agent(configure_sockets_environment, connect_to_sockets
...
+    socket_closed = False
+    cmd_counter = 0
+    while True:
+        cmd_counter += 1
+        response = receiver_sockets[0].receive(size=True).decode()
+        if response == "":
```
Comment (juliamagan) [inline tests/integration/test_wazuh_db/test_wazuh_db.py:166]:
```suggestion
        if response == '':
```

File: tests/integration/test_wazuh_db/test_wazuh_db.py [hunk 18]
```diff
@@ -112,28 +142,32 @@ def test_wazuh_db_create_agent(configure_sockets_environment, connect_to_sockets
...
-
-    # Check get-all-agents chunk limit
-    send_chunk_command('global get-all-agents last_id 0')
-    # Check sync-agent-info-get chunk limit
-    send_chunk_command('global sync-agent-info-get last_id 0')
-    # Check get-agents-by-connection-status chunk limit
-    send_chunk_command('global get-agents-by-connection-status 0 active')
-    # Check disconnect-agents chunk limit
-    send_chunk_command('global disconnect-agents 0 {} syncreq'.format(str(int(time.time()) + 1)))
+        if status != "due":
```
Comment (juliamagan) [inline tests/integration/test_wazuh_db/test_wazuh_db.py:170]:
```suggestion
        if status != 'due':
```

File: tests/integration/test_wazuh_db/test_wazuh_db.py [hunk 19]
```diff
@@ -112,28 +142,32 @@ def test_wazuh_db_create_agent(configure_sockets_environment, connect_to_sockets
...
+            break
+
+    assert socket_closed, f'Socket never closed. Received {cmd_counter} commands. Last command: {response}'
```
Comment (juliamagan) [inline tests/integration/test_wazuh_db/test_wazuh_db.py:173]:
```suggestion
    assert socket_closed, f"Socket never closed. Received {cmd_counter} commands. Last command: {response}"
```

File: tests/integration/test_wazuh_db/test_wazuh_db.py [hunk 20]
```diff
@@ -112,28 +142,32 @@ def test_wazuh_db_create_agent(configure_sockets_environment, connect_to_sockets
...
+    time.sleep(2)
+    socket_closed = False
+    cmd_counter = 0
+    while True:
+        cmd_counter += 1
+        response = receiver_sockets[0].receive(size=True).decode()
+        if response == "":
+            socket_closed = True
+            break
         status = response.split(" ", 1)[0]
```
Comment (juliamagan) [inline tests/integration/test_wazuh_db/test_wazuh_db.py:169]: Same as earlier splits

**Review by juliamagan — CHANGES_REQUESTED** [review]
It is needed to change the quotes in the yaml and the test following the next format:
```
x = f"{var}string"
x = 'raw string'
x = "this is my raw 'string' with single quotation marks"
```
Some of them were requested, but there are more.

File: tests/integration/test_wazuh_db/test_wazuh_db.py [hunk 21]
```diff
@@ -112,28 +142,32 @@ def test_wazuh_db_create_agent(configure_sockets_environment, connect_to_sockets
...
-    # Check get-all-agents chunk limit
-    send_chunk_command('global get-all-agents last_id 0')
-    # Check sync-agent-info-get chunk limit
-    send_chunk_command('global sync-agent-info-get last_id 0')
-    # Check get-agents-by-connection-status chunk limit
-    send_chunk_command('global get-agents-by-connection-status 0 active')
-    # Check disconnect-agents chunk limit
-    send_chunk_command('global disconnect-agents 0 {} syncreq'.format(str(int(time.time()) + 1)))
+        if status != "due":
+            break
```
Comment (juliamagan) [inline tests/integration/test_wazuh_db/test_wazuh_db.py:171]: We could change this while to:

```
    while not socket_closed and status == 'due':
        cmd_counter += 1
        response = receiver_sockets[0].receive(size=True).decode()
        if response == '':
            socket_closed = True
        status = response.split(' ', 1)[0]
```

**Review by juliamagan — CHANGES_REQUESTED** [review]

**Comment by juliamagan** [comment]
After talking with @snaow we have considered that new development or changes into existing tests should follow new repository standards about Wazuh daemons. For now, it is required to use fixtures to start only required daemons, and turn them off when the test is over. That means, that the default status of Wazuh would be off and each test should be responsible for daemons handling.
In case of the need of all Wazuh daemon, the test should disable every module not necessary for the testing in ossec.conf.

The reasons for this change are:

Faster testing
More encapsulated tests

Also, we need to implement the common fixtures for auto-configure local internal options and file monitoring.

**Comment by pereyra-m** [comment]
### Validation

After implementing the changes, the proper tests were run.
**Jenkins**: https://ci.wazuh.info/job/Test_integration/14836/console

| Test Executions | Date  | By  | Status | 
|--|--|--|--|
|[report1.zip](https://github.com/wazuh/wazuh-qa/files/7464793/report1.zip)| 2021-11-02| Matias | :green_circle:
|[report2.zip](https://github.com/wazuh/wazuh-qa/files/7464792/report2.zip)| 2021-11-02| Matias | :green_circle:
|[report3.zip](https://github.com/wazuh/wazuh-qa/files/7464791/report3.zip)| 2021-11-02| Matias | :green_circle:

File: tests/integration/test_wazuh_db/test_wazuh_db.py (hunk 17 above)
Comment (DProvinciani) [reply tests/integration/test_wazuh_db/test_wazuh_db.py]: Fixed

File: tests/integration/test_wazuh_db/test_wazuh_db.py (hunk 18 above)
Comment (DProvinciani) [reply tests/integration/test_wazuh_db/test_wazuh_db.py]: Fixed

File: tests/integration/test_wazuh_db/test_wazuh_db.py (hunk 19 above)
Comment (DProvinciani) [reply tests/integration/test_wazuh_db/test_wazuh_db.py]: Fixed

File: tests/integration/test_wazuh_db/test_wazuh_db.py (hunk 20 above)
Comment (DProvinciani) [reply tests/integration/test_wazuh_db/test_wazuh_db.py]: Fixed

File: tests/integration/test_wazuh_db/test_wazuh_db.py [hunk 22]
```diff
@@ -112,28 +142,32 @@ def test_wazuh_db_create_agent(configure_sockets_environment, connect_to_sockets
...
 
-def test_wazuh_db_chunks(configure_sockets_environment, connect_to_sockets_module, pre_insert_agents):
-    """Check that commands by chunks work properly when agents amount exceed the response maximum size"""
 
-    def send_chunk_command(command):
-        receiver_sockets[0].send(command, size=True)
-        response = receiver_sockets[0].receive(size=True).decode()
 
+
+def test_wazuh_db_timeout(configure_sockets_environment, connect_to_sockets_module,
```
[the remark this answers is not in the document]
Comment (DProvinciani) [reply tests/integration/test_wazuh_db/test_wazuh_db.py]: Fixed

File: tests/integration/test_wazuh_db/test_wazuh_db.py (hunk 21 above)
Comment (DProvinciani) [reply tests/integration/test_wazuh_db/test_wazuh_db.py]: Fixed

File: tests/integration/test_wazuh_db/test_wazuh_db.py (hunk 16 above)
Comment (DProvinciani) [reply tests/integration/test_wazuh_db/test_wazuh_db.py]: Fixed

**Comment by DProvinciani** [comment]
### Validation

We run three times the Vulnerability Detector tests in tier 0.

| Test Executions | Date  | Status | 
|--|--|--|
|[test_vuldet_1.zip](https://github.com/wazuh/wazuh-qa/files/7523350/test_vuldet_1.zip)| 2021-11-11 | :green_circle:
|[test_vuldet_2.zip](https://github.com/wazuh/wazuh-qa/files/7523357/test_vuldet_2.zip)| 2021-11-11 | :green_circle:
|[test_vuldet_3.zip](https://github.com/wazuh/wazuh-qa/files/7523359/test_vuldet_3.zip)| 2021-11-11 | :green_circle:

**Comment by pereyra-m** [comment]
### Validation

The Wazuh-DB tests were run three times

| Test Executions | Date  | By  | Status | 
|--|--|--|--|
|[wazuh_db_1.zip](https://github.com/wazuh/wazuh-qa/files/7523408/wazuh_db_1.zip)| 2021-11-11| Matias | :green_circle: 
|[wazuh_db_2.zip](https://github.com/wazuh/wazuh-qa/files/7523409/wazuh_db_2.zip)| 2021-11-11| Matias | :green_circle:
|[wazuh_db_3.zip](https://github.com/wazuh/wazuh-qa/files/7523411/wazuh_db_3.zip)| 2021-11-11| Matias | :green_circle:

**Comment by palaciosjeremias** [comment]
### Validation
Successful execution of the tests in Jenkins (contains skipped tests)
https://ci.wazuh.info/job/Test_integration/15817/
https://ci.wazuh.info/job/Test_integration/15818/

File: tests/integration/test_wazuh_db/test_wazuh_db.py [hunk 23]
```diff
@@ -301,3 +364,29 @@ def test_wazuh_db_range_checksum(restart_wazuh, configure_sockets_environment, c
...
+    socket_closed = False
+    cmd_counter = 0
+    status = 'due'
+    while not socket_closed and status == 'due':
+        cmd_counter += 1
+        response = receiver_sockets[0].receive(size=True).decode()
+        if response == '':
+            socket_closed = True
+        else:
+            status = response.split()[0]
```
Comment (juliamagan) [inline tests/integration/test_wazuh_db/test_wazuh_db.py:390]: If the socket never closes, the `while` never ends and the test hangs.

**Review by juliamagan — CHANGES_REQUESTED** [review]
We will need to improve these tests in the future moving the callbacks to common libraries, using common fixtures, and fixing the style of strings or imports. However, it is important to fix `test_wazuh_db.py`.

File: docs/tests/integration/test_vulnerability_detector/index.md [hunk 24]
```diff
@@ -74,15 +86,33 @@ Vulnerability Detector generates the alerts from NVD and providers feed.
...
 
 ### Tier 1
 
 #### Test feeds
 
-- **[test_download_feeds](test_feeds/test_download_feeds.md)**: The tests download
-the different feeds (Redhat, Canonical, Debian, Amazon Linux and NVD), import them, and check if the confirmation message appears
-in the logs.
+- **[test_download_feeds](test_feeds/test_download_feed.md)**: The tests download
+the different feeds (Redhat, Canonical, Debian, and NVD), import them, and check if the confirmation message appears
```
[the remark this answers is not in the document]
Comment (juliamagan) [reply docs/tests/integration/test_vulnerability_detector/index.md]: Done in commit 0ba0725

File: mkdocs.yml [hunk 25]
```diff
@@ -94,6 +94,10 @@ nav:
             - Test scan different cves:  tests/integration/test_vulnerability_detector/test_scan_results/test_scan_different_cves.md
             - Test scan nvd feed: tests/integration/test_vulnerability_detector/test_scan_results/test_scan_nvd_feed.md
             - Test scan providers and nvd feed: tests/integration/test_vulnerability_detector/test_scan_results/test_scan_providers_and_nvd_feed.md
+          - Tests SCAN types:
+            - tests/integration/test_vulnerability_detector/test_scan_types/index.md
+            - Test baseline scan type: tests/integration/test_vulnerability_detector/test_scan_types/test_baseline_scan_type.md
+            - Test partial scan type: tests/integration/test_vulnerability_detector/test_scan_types/test_partial_scan_type.md
```
[the remark this answers is not in the document]
Comment (juliamagan) [reply mkdocs.yml]: Done in commit 0ba0725

**Comment by juliamagan** [comment]
## 15/11/2021

We have removed the skip from the next tests to execute them locally:
- `tests/integration/test_vulnerability_detector/test_general_settings/test_general_settings_min_full_scan_interval.py`
- `tests/integration/test_vulnerability_detector/test_scan_results/test_alert_vulnerability_removal.py`
- `tests/integration/test_vulnerability_detector/test_scan_types/test_baseline_scan_type.py`
- `tests/integration/test_vulnerability_detector/test_scan_types/test_partial_scan_type.py`

### Package
| Version | Revision | Link|
|---|---|---|
4.3.0|40301|https://packages-dev.wazuh.com/warehouse/pullrequests/4.3/rpm/var/wazuh-manager-4.3.0-0.commit3799ec1.x86_64.rpm|


## Testing

### tests/integration/test_vulnerability_detector --tier 0

|  OS    | Local     | Notes
|---      |---    |---    |
| PS1   |  [🔴 ](https://github.com/wazuh/wazuh-qa/files/7540109/PS1-1243-tier0-centos-manager.zip)  | Failed due to known issue https://github.com/wazuh/wazuh/issues/9309
| PS2   |  | |
| PS3   |     | |

### tests/integration/test_vulnerability_detector --tier 1

|  OS    | Local     | Notes
|---      |---    |---    |
| PS1   |    | 
| PS2   |  | |
| PS3   |     | |

* * * 

- :green_circle:: All pass
- :yellow_circle:: Some warnings
- :red_circle:: Some errors/fails
- :large_blue_circle:: In progress 

* * * 

## 05/11/2021

I've opened #2185 in order to solve these failures.

**Pull request merged**